python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -249,3 +253,106 @@ def make_airflow_example_dags():
     return make_dagster_definitions_from_airflow_dag_bag(
         dag_bag=dag_bag, use_ephemeral_airflow_db=use_ephemeral_airflow_db
     )
+
+def make_graph_backed_assets_from_airflow_dag(
+    dag,
+    asset_key_to_task_ids=None,
+    connections=None,
+):
+    """Construct a Dagster Assets forr Airflow DAG.
+
+    Args:
+        dag (DAG): The Airflow DAG to compile into a Dagster job
+        asset_key_to_task_ids (Optional[Dict[str, Union[str, set[str]]]]): A mapping from asset keys to task ids
+        connections (List[Connection]): List of Airflow Connections to be created in the Ephemeral
+            Airflow DB
+
+    Returns:
+        AssetDefinition
+    """
+    cron_schedule = dag.normalized_schedule_interval
+
+    job = make_dagster_job_from_airflow_dag(dag, use_ephemeral_airflow_db=True, connections=connections)
+    graph = job._graph_def
+    start_date = dag.start_date if dag.start_date else dag.default_args.get("start_date")
+
+    # if no asset_
+    if asset_key_to_task_ids is None or asset_key_to_task_ids == {}:
+        # leaf nodes have no downstream nodes
+        forward_edges, _ = _create_adjacency_lists(graph.solids, graph.dependency_structure)
```

Comment:
Ideally we'd avoid using the legacy `solids`.  We could do that either by adding a `nodes` property to `GraphDefinition` or using `graph.nodes_dict.values()`.